src/internal/race-to.mjs

```diff
@@ -0,0 +1,23 @@
+export default function raceTo (predicate, notFoundValue, promises) {
+  return new Promise(resolve => {
+    let promiseCount = 0
+    let settledPromiseCount = 0
+    let resolved = false
+    for (const promise of promises) {
+      promise
```

Comment:
Instead of looping through them, you can wrap all promises in a promise race.

Reply:
> I don't want to resolve if an input promise rejects

maybe you should let this blown up if it rejects

And you lost me with the rest of explanation :-)

Maybe you can just go on implementing the test, and then I can fiddle with it. So I will either understand how it works or propose an alternative

Reply:
I just meant that you can't say that `canTakeAny` is true if all you do is race to the first iterable that is ready to tell you that `canTake: false`. I probably should make sure it blows up in canTake ever actually rejects though.